docs/diagnostics/PX1034.md

```diff
@@ -0,0 +1,50 @@
+# PX1034
+This document describes the PX1034 diagnostic.
+
+## Summary
+
+| Code   | Short Description                                 | Type  | Code Fix    | 
+| ------ | ------------------------------------------------- | ----- | ----------- | 
+| PX1034 | The DAC does not have an explicit foreign key declaration | Warning (ISV Level 3: Informational) | Unavailable | 
+
+## Diagnostic Description
+You can define foreign key in DACs. The diagnostic suggests adding foreign key definition to a DAC. The warning is shown only for DACs that have either PXCacheName or PXPrimaryGraph attribute.
```

Comment:
There is a couple of other restrictions. Diagnostic will not check abstract DACs and DACs which has only unbound DAC properties (so-called "virtual" DACs, used for custom popups and filters for inquiry screens)

Reply:
Code fix should be described. For this diagnostic code fix will add public static class FK.
Inside it will place a big comment with examples of different foreign keys and a list of DAC properties which have a high probability of holding foreign key. This is determined by the analysis of attributes declared on DAC property. We look for the following attribute or attributes derived from them:
- PXParent 
- PXDBDefault 
- PXSelector 
- PXDimensionSelector 
- PXForeignReference